app/src/main/java/com/example/taskyy/ui/LoginScreen.kt

```diff
@@ -56,49 +60,57 @@ fun LoginScreen(state: LoginState, onEvent: (LoginEvent) -> Unit) {
                     .fillMaxHeight(.75f)
                     .padding(30.dp)
             ) {
-                CreateUserNameField(state, onEvent)
+                CreateEmailField(state, onEvent)
                 Spacer(modifier = Modifier.height(10.dp))
                 CreatePasswordField(state, onEvent)
                 Spacer(modifier = Modifier.height(10.dp))
-                CreateLoginButton()
+                CreateLoginButton(state, onEvent, navController, stringResource(R.string.login))
                 Row(
                 ) {
                     CreateBottomText()
-                    CreateSignUpLink()
+                    CreateSignUpLink(navController)
                 }
             }
         }
     }
 }
 
 @Composable
-private fun WelcomeBackText() {
+fun TopText(text: String) {
     Text(
-        text = "Welcome Back!",
+        text = text,
         color = Color.White,
         fontSize = 30.sp,
         fontWeight = FontWeight.Bold
     )
 }
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun CreateUserNameField(state: LoginState, onEvent: (LoginEvent) -> Unit) {
+fun CreateEmailField(state: LoginState, onEvent: (LoginEvent) -> Unit) {
```

Comment:
To answer your question:
You wouldn't want to make this field dependent on `LoginState` since as you realized this won't make it reusable for screens where you don't have a `LoginState`. Instead, only give this screen the fields it needs. For a text field, that might be the text itself, whether the field is valid and maybe a hint.
That way, you can also call this something more generic, like `TaskyTextField` which just styles it based on the mockups, but allows you to use it for different types of text fields

Reply:
If I were to not include that state here, how would I update the onValueChange? I use that to update the email in the view model.

Reply:
Not 100% sure what you mean, but you'd just pass an `onValueChange: (String) -> Unit` lambda to this text field. In your screen, you can then do this:
```
CreateEmailField(
    text = state.text,
    onValueChange = { onEvent(LoginEvent.EmailTextChanged(it) },
    ...
)
```

Reply:
In your example your using state.text to set the text value. But you don't have state so when the user updates the email field, if I don't have the state, how do I update the textfield?

Reply:
Ok I see how to not pass the state but what about specific events? It accepts a LoginEvent to the LoginEvent.OnEmailChanged(it) method.

Reply:
For each event, you simply pass a lambda just like when the text changed. For example, this could also be done:
```
CreateEmailField(
    text: String,
    onValueChange: (String) -> Unit,
    onEndIconClick: () -> Unit
)
```
Unless you have a component that really needs most of the events declared in the screen event class (here `LoginEvent`), I'd prefer using separate lambdas to make this composable independent of the screen and reusable therefore

Reply:
I'm sorry but I'm not understanding this. If I don't pass an event into the composable, how would onEvent in the view model know what to do

Reply:
Your composable will look like this:
```
@Composable
CreateEmailField(
    text: String,
    onValueChange: (String) -> Unit,
)
```
Then, you add this to your `LoginScreen` for example (which has the `onEvent` lambda):
```
// Content before
CreateEmailField(
    text = state.text,
    onValueChange = { email ->
        onEvent(LoginEvent.OnEmailChange(email)
    }
)
```
The onEvent lambda still exists, you just don't add it to the CreateEmailField composable, since all you care about is just one or two evens (like when the text changes)

Reply:
When I try to use the onEvent lambda, it doesn't recognize it if I don't pass it into the composable. It says that onEvent is an unresolved reference. It also says I'm not using the onValueChange parameter that I'm passing in. I must be doing something incorrectly

Reply:
Can you post the code as you have it right now? The whole screen Composable as well as the whole text field composable